### DISCUSSION TITLE
Facing "Graceful stop of task failed" error for CosmosDBSourceConnector.

### DISCUSSION THREAD

**Discussion by ggulat01** [microsoft/kafka-connect-cosmosdb#543] [Q&A]
I am facing  "Graceful stop of task failed" error on updating configuration for CosmosDBSourceConnector, even while resuming connector and on restart on kafka connect.
I have also increased "task.shutdown.graceful.timeout.ms" to 30000 but this has not solved this issue.
In CosmosDBSourceConnector configuration, I have set "connect.cosmos.task.poll.interval" property value to 100, other properties are using default value.
It seems to me this issue comes only when there are changes available to read and there comes a request to stop task.
It would be great if someone will be able to help me out with this, as after this error,  CosmosDBSourceConnector stops reading data from the source.

Thanks

**Comment by kushagraThapar** [comment]
@xinlian12 can you please take a look at this, thanks!

**Reply by ggulat01** [reply]
@xinlian12, I am looking forward to your update.
Please let me know if you need any other information.

Thanks

**Comment by xinlian12** [comment]
@ggulat01  sorry for the delay, will take a look of the issue and update soon, thanks

**Comment by xinlian12** [comment]
@ggulat01 I believe this issue is related to what we are trying to fix in this PR: https://github.com/microsoft/kafka-connect-cosmosdb/pull/545. 

We plan to release the fix by the end of this week or early next week, will update here once the new library is there. Thanks

**Reply by ggulat01** [reply]
@kushagraThapar @xinlian12 Thank you for providing the fix, appreciate it!

**Reply by xinlian12** [reply]
@ggulat01 the fix has been released in 1.13.0 - https://github.com/microsoft/kafka-connect-cosmosdb/releases/tag/v1.13.0. Can you try and see whether it solved the issue you have observed? thanks

**Reply by ggulat01** [reply]
@xinlian12 I am not facing this issue after upgrading its version to 1.13.0. 
Thanks for the quick fix.

**Reply by xinlian12** [reply]
thanks @ggulat01
